src/github.com/matrix-org/dendrite/roomserver/types/types.go

```diff
@@ -1,10 +1,82 @@
 // Package types provides the types that are used internally within the roomserver.
 package types
 
+import (
+	"github.com/matrix-org/gomatrixserverlib"
+)
+
 // A PartitionOffset is the offset into a partition of the input log.
 type PartitionOffset struct {
 	// The ID of the partition.
 	Partition int32
 	// The offset into the partition.
 	Offset int64
 }
+
+// A StateKey is a pair of a numeric event type and a numeric state key.
+// It is used to lookup state entries.
+type StateKey struct {
```

Comment:
Naming: If I see a function:
```go
func foo(a StateKey) {

}
```

I am *always* going to think `a` is `event.state_key`, no matter how much you doc otherwise. Can we call this something else, like `StateTuple`, `StateKeyTuple`, `StateTupleNID`, etc?